src/gc/impl/conservative/gc.d

```diff
@@ -292,6 +297,70 @@ struct GC
             gcx.reserve(config.initReserve << 20);
         if (config.disable)
             gcx.disabled++;
+
+        initProxy();
+        gc_setProxy(&pthis);
+
+    }
+
+    private void initProxy()
+    {
+        pthis.gc_enable = function void() { gcInstance.enable();};
+        pthis.gc_disable = function void() { gcInstance.disable();};
+        pthis.gc_term = function void() {
+        gcInstance.fullCollectNoStack(); // not really a 'collect all' -- still scans
+                                  // static data area, roots, and ranges.
+        gcInstance.Dtor();
+        };
+
+        pthis.gc_collect = function void() { gcInstance.fullCollect();};
+        pthis.gc_minimize = function void() { gcInstance.minimize();};
+
+        pthis.gc_getAttr = function uint(void* p) { return gcInstance.getAttr(p);};
+        pthis.gc_setAttr = function uint(void* p, uint a) { return gcInstance.setAttr(p, a);};
+        pthis.gc_clrAttr = function uint(void* p, uint a) { return gcInstance.clrAttr(p, a);};
+
+        pthis.gc_malloc = function void*(size_t sz, uint ba, const TypeInfo ti) { return gcInstance.malloc( sz, ba, null, ti );};
+        pthis.gc_qalloc = function BlkInfo(size_t sz, uint ba, const TypeInfo ti) {
+            BlkInfo retval;
+            retval.base = gcInstance.malloc( sz, ba, &retval.size, ti );
+            retval.attr = ba;
+            return retval;};
+        pthis.gc_calloc = function void*(size_t sz, uint ba, const TypeInfo ti) { return gcInstance.calloc( sz, ba, null, ti );};
+        pthis.gc_realloc = function void*(void* p, size_t sz, uint ba, const TypeInfo ti) { return gcInstance.realloc( p, sz, ba, null, ti );};
+        pthis.gc_extend = function size_t(void* p, size_t mx, size_t sz, const TypeInfo ti) { return gcInstance.extend( p, mx, sz, ti );};
+        pthis.gc_reserve = function size_t(size_t sz) { return gcInstance.reserve(sz);};
+        pthis.gc_free = function void(void* p){ gcInstance.free(p);};
+
+        pthis.gc_addrOf = function void*(void* p) { return gcInstance.addrOf(p);};
+        pthis.gc_sizeOf = function size_t(void* p) { return gcInstance.sizeOf(p);};
+
+        pthis.gc_query = function BlkInfo(void* p) { return gcInstance.query(p);};
+        pthis.gc_stats = function GCStats() {GCStats stats = void; gcInstance.getStats( stats ); return stats;};
+
+        pthis.gc_addRoot = function void(void* p) { gcInstance.addRoot(p);};
+        pthis.gc_addRange = function void(void* p, size_t sz, const TypeInfo ti) { gcInstance.addRange( p, sz, ti );};
+
+        pthis.gc_removeRoot = function void(void* p) { gcInstance.removeRoot(p);};
+        pthis.gc_removeRange = function void(void*p) { gcInstance.removeRange(p);};
+        pthis.gc_runFinalizers = function void(in void[] segment) { gcInstance.runFinalizers(segment);};
+
+        pthis.gc_inFinalizer = function bool() { return gcInstance.inFinalizer;};
+
+        pthis.gc_setProxy = function void(Proxy* p) {
```

Comment:
Let's rather add the `rootIter` and `rangeIter` interfaces than this confusing `gc_setProxy` stuff.